src/System/Process/Typed.hs

```diff
@@ -624,7 +618,17 @@ startProcess pConfig'@ProcessConfig {..} = liftIO $ do
 
     pExitCode <- newEmptyTMVarIO
     waitingThread <- async $ do
-        ec <- P.waitForProcess pHandle
+        ec <-
+          if multiThreadedRuntime
+            then P.waitForProcess pHandle
+            else
+              let loop = do
+                    threadDelay 20000
```

Comment:
Could we make the the loop exponentially increasing up to the limit of the `20000`, so that children that terminate quickly are reaped as quick as possible?

Reply:
Is 20,000 the magic number you'd mentioned from the run time system? Is it reasonable to hard-code it here, or should we be using some kind of variable exposed by GHC?

Reply:
Yes, 20 ms, [the default for `-V`](https://downloads.haskell.org/~ghc/8.4.3/docs/html/users_guide/profiling.html#rts-flag--V%20%E2%9F%A8secs%E2%9F%A9):

>  `-V ⟨secs⟩`
>     Default:  0.02
>
>    Sets the interval that the RTS clock ticks at, which is also the sampling interval of the time and allocation profile. The default is 0.02 seconds.

I think you can query this time with [`getConcFlags`](https://hackage.haskell.org/package/base-4.11.1.0/docs/GHC-RTS-Flags.html#v:getConcFlags):

```
Prelude> import GHC.RTS.Flags 
Prelude GHC.RTS.Flags> getConcFlags 
ConcFlags {ctxtSwitchTime = 20000000, ctxtSwitchTicks = 2}
```

`ctxtSwitchTime` has it in nanoseconds.